lib/dirac_domain_wall.cpp

```diff
@@ -33,46 +33,43 @@ namespace quda {
     return *this;
   }
 
-  void DiracDomainWall::checkDWF(const ColorSpinorField &out, const ColorSpinorField &in) const
+  void DiracDomainWall::checkDWF(cvector_ref<const ColorSpinorField> &out, cvector_ref<const ColorSpinorField> &in) const
```

Comment:
Slightly orthogonal to the PR, but I've noticed over the course of this review that there are several _nearly_ identical `checkDWF` functions across the various flavors of DWF/Mobius. The one semi-outlier is Mobius having a separate check for `zMobius`, but even that seems sufficiently redundant of the non-zMobius path. 

I think we should write one `checkDWF` that can be inserted in the base `Dirac` class or something of the sort and reused across all of the 5-d operators.

Reply:
[053bfaa](https://github.com/lattice/quda/pull/1465/commits/053bfaa286655ac83c057583e3c70ee3688c9f82)